packages/core/src/sync-realtime/service.ts

```diff
@@ -619,6 +619,9 @@ export class RealtimeSyncService extends Emittery<RealtimeSyncEvents> {
    * @returns True if a re-org has occurred.
    */
   reconcileReorg = async (latestBlockNumber: number) => {
+    const hasDeepReorg = await this.reconcileDeepReorg(latestBlockNumber);
+    if (hasDeepReorg) return true;
+
     this.common.logger.debug({
       service: "realtime",
       msg: `Checking logs for inconsistencies for blocks from ${
```

Comment:
Maybe `Validating local chain from block ${this.finalizedBlock.number + 1} to ${latestBlockNumber} (network=${this.network.name})`